File: BackEnd/Api/Services/Models/Topping.cs
using System;
using System.Collections.Generic;
using DAO = Api.Data;

namespace Api.Services.Models
{
    public class Topping
    {
        public int ToppingId { get; set; }
        public string ToppingName { get; set; }
        public int ToppingQuantity { get; set; }
        public bool Checked { get; set; }
        //public ICollection<PizzaTopping> PizzaToppings { get; set; }

        //toDAO
        public static DAO.Topping toDAO(Topping dto)
        {
            var dao = new DAO.Topping();
            dao.ToppingId = dto.ToppingId;
            dao.ToppingName = dto.ToppingName;
            dao.ToppingQuantity = dto.ToppingQuantity;
            return dao;
        }

        //fromDAO
        public static Topping fromDAO(DAO.Topping dao)
        {
            var dto = new Topping();
            dto.ToppingId = dao.ToppingId;
            dto.ToppingName = dao.ToppingName;
            dto.ToppingQuantity = dao.ToppingQuantity;
            return dto;
        }
    }
}

File: FrontEnd/src/app/models/topping.ts
export class Topping{
    public toppingIdNumber: string;
    public toppingName: string;
    public toppingQuantity: string;
    public toppingPrice: string;

    constructor(toppingIdNumber: string, toppingName: string, toppingQuantity: string, toppingPrice: string)
    {
        this.toppingIdNumber = toppingIdNumber;
        this.toppingName = toppingName;
        this.toppingQuantity = toppingQuantity;
        this.toppingPrice = toppingPrice;
    }
}